src/main/java/com/joss/conductor/mobile/Locomotive.java

```diff
@@ -163,9 +168,45 @@ private IOSDeviceUtil iosDeviceUtil() {
         return iosDeviceUtil == null ? IOSDeviceUtil.getInstance() : iosDeviceUtil;
     }
 
+    @Rule
+    public TestRule watchman = new TestWatcher() {
+        boolean failure;
+        Throwable e;
+        Description description;
+
+
+
+        @Override
+        protected void failed(Throwable e, Description description) {
+            if (configuration.screenshotsOnFail()) {
+                failure = true;
+                this.e = e;
+                this.description = description;
+            }
+        }
+
+        /**
+         * Take screenshot if the test failed.
+         */
+        @Override
+        protected void finished(Description description) {
+            super.finished(description);
+            if (configuration.screenshotsOnFail()) {
+                if (failure) {
+                    ScreenShotUtil.take(Locomotive.this,
+                            description.getDisplayName(),
+                            e.getMessage());
+                }
+                Locomotive.this.driver.quit();
```

Comment:
Yes, if the featured is enabled `finished` is in charge of quitting the driver, regardless of whether the test failed or not.